toolchain/mfc/run/run.py

```diff
@@ -24,30 +31,67 @@ def validate_job_options() -> None:
             raise MFCException(f'RUN: {ARG("email")} is not a valid e-mail address.')
 
 
-def run_targets(targets: typing.List[MFCTarget]):
-    cons.print("[bold]Run[/bold]")
-    cons.indent()
+def __profiler_prepend() -> typing.List[str]:
+    if ARG("ncu") is not None:
+        if not does_command_exist("ncu"):
+            raise MFCException("Failed to locate [bold green]NVIDIA Nsight Compute[/bold green] (ncu).")
 
-    if len(targets) == 0:
-        cons.print(f"> No target selected.")
-        return
+        return ["ncu", "--nvtx", "--mode=launch-and-attach",
+                       "--cache-control=none", "--clock-control=none"] + ARG("ncu")
 
-    input_file = input.load()
+    if ARG("nsys") is not None:
+        if not does_command_exist("nsys"):
+            raise MFCException("Failed to locate [bold green]NVIDIA Nsight Systems[/bold green] (nsys).")
 
-    engine = engines.get_engine(ARG("engine"))
-    engine.init(input_file)
+        return ["nsys", "profile", "--stats=true", "--trace=mpi,nvtx,openacc"] + ARG("nsys")
```

Comment:
i think i had a comment on your original PR that said that nsys profile needs to go in the opposite order against `mpirun` for whatever we do currently (for mpi jobs at least). does this fix that?

Reply:
Noted. I just switched the order.